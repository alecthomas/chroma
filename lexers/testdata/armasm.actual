@ Hello World in ARM Assembly for Linux system

.global _start

_start:
    mov  r7, #4          @ Setup service call 4 (write)
    mov  r0, #1          @ param 1 - File descriptor 1 = stdout
    ldr  r1, =hello      @ param 2 - address of string to print
    mov  r2, #13         @ param 3 - length of hello world string
    svc  0               @ ask linux to write to stdout

    mov  r7, #1          @ Setup service call 1 (exit)
    mov  r0, #0          @ param 1 - 0 = normal exit
    svc  0               @ ask linux to terminate us

.data
hello:    .ascii    "Hello World!\n"