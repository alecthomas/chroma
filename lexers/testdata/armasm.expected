[
  {"type":"CommentSingle","value":"@ Hello World in ARM Assembly for Linux system\n"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordNamespace","value":".global"},
  {"type":"NameLabel","value":" _start\n"},
  {"type":"Text","value":"\n"},
  {"type":"NameLabel","value":"_start"},
  {"type":"Punctuation","value":":"},
  {"type":"Text","value":"\n    mov  "},
  {"type":"NameClass","value":"r7"},
  {"type":"Text","value":", #"},
  {"type":"LiteralNumberInteger","value":"4"},
  {"type":"Text","value":"          "},
  {"type":"CommentSingle","value":"@ Setup service call 4 (write)\n"},
  {"type":"Text","value":"    mov  "},
  {"type":"NameClass","value":"r0"},
  {"type":"Text","value":", #"},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Text","value":"          "},
  {"type":"CommentSingle","value":"@ param 1 - File descriptor 1 = stdout\n"},
  {"type":"Text","value":"    ldr  "},
  {"type":"NameClass","value":"r1"},
  {"type":"Text","value":", ="},
  {"type":"NameLabel","value":"hello"},
  {"type":"Text","value":"      "},
  {"type":"CommentSingle","value":"@ param 2 - address of string to print\n"},
  {"type":"Text","value":"    mov  "},
  {"type":"NameClass","value":"r2"},
  {"type":"Text","value":", #"},
  {"type":"LiteralNumberInteger","value":"13"},
  {"type":"Text","value":"         "},
  {"type":"CommentSingle","value":"@ param 3 - length of hello world string\n"},
  {"type":"Text","value":"    "},
  {"type":"NameNamespace","value":"svc  0"},
  {"type":"Text","value":"               "},
  {"type":"CommentSingle","value":"@ ask linux to write to stdout\n"},
  {"type":"Text","value":"\n    mov  "},
  {"type":"NameClass","value":"r7"},
  {"type":"Text","value":", #"},
  {"type":"LiteralNumberInteger","value":"1"},
  {"type":"Text","value":"          "},
  {"type":"CommentSingle","value":"@ Setup service call 1 (exit)\n"},
  {"type":"Text","value":"    mov  "},
  {"type":"NameClass","value":"r0"},
  {"type":"Text","value":", #"},
  {"type":"LiteralNumberInteger","value":"0"},
  {"type":"Text","value":"          "},
  {"type":"CommentSingle","value":"@ param 1 - 0 = normal exit\n"},
  {"type":"Text","value":"    "},
  {"type":"NameNamespace","value":"svc  0"},
  {"type":"Text","value":"               "},
  {"type":"CommentSingle","value":"@ ask linux to terminate us\n"},
  {"type":"Text","value":"\n"},
  {"type":"KeywordNamespace","value":".data"},
  {"type":"Text","value":"\n"},
  {"type":"NameLabel","value":"hello"},
  {"type":"Punctuation","value":":"},
  {"type":"KeywordNamespace","value":"    .ascii    "},
  {"type":"Punctuation","value":"\""},
  {"type":"LiteralStringDouble","value":"Hello World!\\n"},
  {"type":"Punctuation","value":"\""}
]
